app/sesame/automod/linkcheck/unshorten_test.go

```
@@ -0,0 +1,174 @@
+			t.Errorf("guardDial(%q) allowed, want refused", tt.addr)
+		}
+	}
+}
+
+func TestGuardDialBlocksLiveLoopbackDial(t *testing.T) {
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {}))
+	defer srv.Close()
+
+	e := NewExpander(nil, []string{hostOf(tok(srv.URL))}) // default client carries guardDial
+	req, _ := http.NewRequestWithContext(context.Background(), http.MethodGet, srv.URL, nil)
+	if _, err := e.client.Do(req); err == nil {
```

golangci-lint (2.12.2)

[error] response body must be closed

(bodyclose)